code/datums/elements/eldritch_curse.dm

```diff
@@ -0,0 +1,41 @@
+/// -- NEW Wiskey's Heretic Curse --
+
+/datum/element/eldritch_curse
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
+	id_arg_index = 2
+	var/duration = 5 MINUTES
+	var/datum/callback/curse_proc
+	var/datum/callback/uncurse_proc
+
+/datum/element/eldritch_curse/Attach(datum/target, nduration, datum/callback/curse, datum/callback/uncurse)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.duration = nduration
+	src.curse_proc = curse
+	src.uncurse_proc = uncurse
+	RegisterSignal(target, COMSIG_ITEM_PICKUP, .proc/pickup_safety_check)
+
+/datum/element/eldritch_curse/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, list(COMSIG_ITEM_PICKUP))
+
+/datum/element/eldritch_curse/proc/pickup_safety_check(datum/source, mob/user)
+	SIGNAL_HANDLER
+
+	if(curse_safety_check(source, user))
+		return
+		
+	curse_proc.Invoke(user)
+	addtimer(CALLBACK(src, curse_proc/uncurse_proc, user),duration)
+	src.Detach(source)
+
+/datum/element/eldritch_curse/proc/curse_safety_check(datum/source, mob/living/carbon/user)
```

Comment:
It will maybe had sense if it would check is the dude picking up the item a heretic/heretic monster or no